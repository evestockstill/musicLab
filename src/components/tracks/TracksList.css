@import '../App.css';

.LinkTracks {
  text-align: center;
  color: inherit;
  text-decoration: inherit;
  margin: 0 auto;
  padding: 0.8rem 0.3rem 1rem 0.3rem;
  .tracks_list {
    font-size: 2rem;
    text-align: center;
    display: inline-block;
    margin: .5rem 1rem;
    padding: .5rem .7rem;
    border-width: 2px;
    border-style: solid;
    border-image: 
    linear-gradient(
      to bottom,
      $orange, 
      rgba(0, 0, 0, 0)
    ) 1 100%;
    
  }
}
.TrackNames {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    margin: 2.5rem;
    padding: 1rem;
}
.spinner {
  width: 40px;
  height: 40px;

  position: relative;
  margin: 100px auto;
}

.bounce1, .bounce2 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: #333;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  
  -webkit-animation: sk-bounce 2.0s infinite ease-in-out;
  animation: sk-bounce 2.0s infinite ease-in-out;
}

.bounce2 {
  -webkit-animation-delay: -1.0s;
  animation-delay: -1.0s;
}

@-webkit-keyframes sk-bounce {
  0%, 100% { -webkit-transform: scale(0.0) }
  50% { -webkit-transform: scale(1.0) }
}

@keyframes sk-bounce {
  0%, 100% { 
    transform: scale(0.0);
    -webkit-transform: scale(0.0);
  } 50% { 
    transform: scale(1.0);
    -webkit-transform: scale(1.0);
  }
}
.title {
    font-size: 3.8rem;
    padding-top: 2rem;
    margin: 1rem 2rem;
    text-align: center;
    font-family: 'Permanent Marker', cursive;
}