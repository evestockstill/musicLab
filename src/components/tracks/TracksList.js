import React from 'react';
import { useTracks } from '../../hooks/tracks';
const TracksList = ({ match }) => {

  const tracks = useTracks(match.params.albumId);
  const listOfTracks = tracks.map(track => {
    console.log(track);
    return (
      <li key={track.id}>{track.title}</li>
    );
  });
  return (
    <>
      <h1>Tracks</h1>
      <ul>
        {listOfTracks}
      </ul>
    </>
  );
};

export default TracksList;
