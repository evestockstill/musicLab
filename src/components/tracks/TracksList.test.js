import React from 'react';
import { shallow } from 'enzyme';
import TracksList from './TracksList';

describe('TracksList component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<TracksList match={{ params: { id: '1' }
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
