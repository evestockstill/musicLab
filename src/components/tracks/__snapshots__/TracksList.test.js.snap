// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TracksList component matches a snapshot 1`] = `
<h1>
  Tracks
</h1>
`;
